spec/services/search/postgres/comment_spec.rb

```diff
@@ -0,0 +1,118 @@
+require "rails_helper"
+
+RSpec.describe Search::Postgres::Comment, type: :service do
+  let(:comment) { create(:comment) }
+
+  describe "::search_documents" do
+    context "when filtering Commentables" do
+      it "does not include comments from Articles that are unpublished", :aggregate_failures do
+        comment_text = "Ruby on Rails rocks!"
+        published_article = create(:article, title: "Published Article", published: true)
+        unpublished_article = create(:article, title: "Unpublished Article", published: true)
+        comment_on_published_article = create(:comment, body_markdown: comment_text, commentable: published_article)
+        comment_on_unpublished_article = create(:comment, body_markdown: comment_text, commentable: unpublished_article)
+        unpublished_article.update_columns(published: false)
```

Comment:
Was the creation of a `published` "unpublished" article with a subsequent update to unpublish it intentional?

Reply:
Yes - it mimics the behavior of publishing an article and then unpublishing it. And this approach avoids having to skip validations because we can't `create` an `Article` as unpublished.

Reply:
Sorry, I should clarify, we can't create a comment on an unpublished article. And here we create the article first so trying to do 

```ruby
comment_on_unpublished_article = create(:comment, body_markdown: comment_text, commentable: unpublished_article)
```

fails validations

https://github.com/forem/forem/blob/cf0baf05f918ade81c5f69132b8b9ce330cc4140/app/models/comment.rb#L49

https://github.com/forem/forem/blob/cf0baf05f918ade81c5f69132b8b9ce330cc4140/app/models/comment.rb#L330-L332